src/Auth/authModule.js

```diff
@@ -0,0 +1,22 @@
+import { createAction, handleActions } from 'redux-actions';
+import localStorage from '../store/localStorage';
+
+export const CHANGE_USERNAME = 'CHANGE_USERNAME';
+export const LOGIN_REQUEST = 'LOGIN_REQUEST';
+
+export const changeUsername = createAction(CHANGE_USERNAME);
+export const loginRequest = createAction(LOGIN_REQUEST);
+
+const initialState = {
+  username: localStorage.getItem('KIZUNA_USERNAME') || '',
+};
+
+export const reducer = handleActions({
+  [CHANGE_USERNAME]: (state = initialState, { payload }) => {
+    const { username } = payload;
+    return Object.assign({}, state, {
+      username,
+    });
```

Comment:
Wait, I didn't mean to ask you to use stage-2 though. I suggested that because you used it first some else.